scripts/parallel.py

```diff
@@ -139,30 +139,33 @@ def main(cfg: DictConfig):
     # For now, all configs are the same
     envs.call("update_config", env_cfg, reset=False)
     # Need separate seeds for each environment
-    rkey = utils.JaxRKey(seed)
-    seeds = [rkey.next_seed() for _ in range(world_draws)]
-
+    seeds = [seed] * world_draws
+    observations, infos = envs_wrapper.reset(seed=seeds)
+    
+    mc_seeds = [random.randint(0, 2**32 - 1) for _ in range(world_draws)]
     # Providing a single seed or a list of seeds both produce variable world inits
     # If you wanted duplicated worlds, provide a list of the same seed [seed]*world_draws
     observations, infos = envs_wrapper.reset(seed=seeds)
 
     for step in tqdm(range(duration), desc="Step"):
         if step >= warmup_steps:
+            envs.call("set_mc_seeds", mc_seeds)
             # Get our vectorized losses
             # Each item is a tuple of (losses, log_probs, collisions)
-            result = envs_wrapper.unwrapped.call("simulate_mc")
+            result = envs_wrapper.unwrapped.call("simulate_mc", n_sims=env_cfg['n_montecarlo'] // world_draws)
```

Comment:
You should check out what `call` does in `AsyncVectorEnv` and why this entails a following modification elsewhere in the code.